### ISSUE TITLE
Online tiles do not download after changing Data storage folder

### ISSUE THREAD

**Issue by janvlug** [osmandapp/OsmAnd#17737]
## Description
My device memory was full, and I have a big micro-sd. So I decided to change the Data storage folder for OsmAnd.
However, the online tiles from Openstreetmap are now not downloading any more. I keep seeing the blurred tiles when I zoom in.

### How to reproduce
Change the Data storage folder to one of:
External storage 1
Multiuser storage 1

### Actual result
The online tiles are not downloaded any more.

### Expected result
Online tiles should be downloaded when needed.

It is strange that sometimes, after changing the settings several times, the online tiles download again, usually only for a short period of time. I switched back now to internal memory, but there are now downloaded maps.

**Environment**
OsmAnd Version: 4.4.7
Android/iOS version: Android 11
Device model: Nokia 4.2

**Comment by yuriiurshuliak** [comment]
Attempted to replicate the issue by following the aforementioned steps. Specifically, changed the data storage folder for OsmAnd and promptly downloaded Microsoft Hybrid online tiles on a larger zoom. As a result, the 104 MB map was successfully downloaded. Kindly furnish a comprehensive account of the reproduction steps along with their outcomes. Additionally, kindly supply screenshots and videos showcasing this bug. And do not forget that the zoom for downloading must be adjusted:
https://osmand.net/docs/user/map/raster-maps#download--update-tiles

OsmAnd~ 4.6.0#497m, released: 2023-07-28

<table style="width:100%">
  <tr>
    <td> <video source src="https://github.com/osmandapp/OsmAnd/assets/127092082/0fca1420-4e36-440b-8288-f3a9f91ca6fa"  width="300" height="auto"/>   </td>
     <th><img src="https://github.com/osmandapp/OsmAnd/assets/127092082/4c3595b6-9832-41f3-a716-a69bff825226"  width="300" height="auto"/> </th>
  </tr>
</table>

**Comment by janvlug** [comment]
It is difficult to exactly provide reproducible steps. Because the behavior seems not really reproducible.
Maybe somehow my sd-card storage is not working well or mis-configured.
But what I notice now is:
I have "Internal app memory" selected. With 44.8 GB free (of 59.46 GB).
When I set a navigation destination, I see a "Download maps" sentence where I can click to download.
There I click to do the suggested download and I see: "Downloading Serbia Standard Map" and the progress bar proceeds until ~100%. Now I'm trying to reproduce I saw only very briefly an IO error appearing. But other times it seemed to work, but the map never showed up in the Downloaded maps. With internal memory used, online tiles download well.
When I set External storage 1 or Multiuser storage 1, I do see the downloaded maps, but online tiles do not download.

When I try to take screenshots, I get: "Couldn't save screenshot Taking screenshots isn't allowed by the app or your organization". Maybe this also indicates an issue with saving files.
